File: cariro_implant_academy/lib/Models/MedicalModels/TreatmentPlanModel.dart
import 'package:cariro_implant_academy/Models/DTOs/DropDownDTO.dart';

import '../ImplantModel.dart';

class TreatmentPlanModel {
  int? id;
  int? patientId;
  int? operatorId;
  DropDownDTO? operator;
  String? date;
  List<TreatmentPlanSubModel>? treatmentPlan;

  TreatmentPlanModel({this.id, this.patientId, this.operatorId, this.operator, this.date, this.treatmentPlan});

  TreatmentPlanModel.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    patientId = json['patientId'];
    operatorId = json['operatorId'];
    operator = json['operator'] != null ? new DropDownDTO.fromJson(json['operator']) : null;
    date = json['date'];
    treatmentPlan = ((json['treatmentPlan'] ?? []) as List<dynamic>).map((e) => TreatmentPlanSubModel.fromJson(e)).toList();
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['patientId'] = this.patientId;
    // data['operatorId'] = this.operatorId;
    if (this.operator != null) {
      //data['operator'] = this.operator!.toJson();
    }
    data['date'] = this.date;
    data['treatmentPlan'] = (this.treatmentPlan ?? <TreatmentPlanSubModel>[]).map((e) => e.toJson()).toList();
    return data;
  }
}

class TreatmentPlanSubModel {
  int? id;
  int? patientId;
  DropDownDTO? patient;
  int? tooth;
  TreatmentPlanFieldsModel? rootCanalTreatment;
  TreatmentPlanFieldsModel? restoration;
  TreatmentPlanFieldsModel? pontic;
  TreatmentPlanFieldsModel? extraction;
  TreatmentPlanFieldsModel? simpleImplant;
  TreatmentPlanFieldsModel? immediateImplant;
  TreatmentPlanFieldsModel? expansionWithImplant;
  TreatmentPlanFieldsModel? splittingWithImplant;
  TreatmentPlanFieldsModel? gbrWithImplant;
  TreatmentPlanFieldsModel? openSinusWithImplant;
  TreatmentPlanFieldsModel? closedSinusWithImplant;
  TreatmentPlanFieldsModel? guidedImplant;
  TreatmentPlanFieldsModel? expansionWithoutImplant;
  TreatmentPlanFieldsModel? splittingWithoutImplant;
  TreatmentPlanFieldsModel? gbrWithoutImplant;
  TreatmentPlanFieldsModel? openSinusWithoutImplant;
  TreatmentPlanFieldsModel? closedSinusWithoutImplant;
  TreatmentPlanFieldsModel? scaling;
  TreatmentPlanFieldsModel? crown;

  TreatmentPlanSubModel(
      {this.id,
      this.patientId,
      this.patient,
      this.tooth,
      this.rootCanalTreatment,
      this.restoration,
      this.pontic,
      this.extraction,
      this.simpleImplant,
      this.immediateImplant,
      this.expansionWithImplant,
      this.splittingWithImplant,
      this.gbrWithImplant,
      this.openSinusWithImplant,
      this.closedSinusWithImplant,
      this.guidedImplant,
      this.expansionWithoutImplant,
      this.splittingWithoutImplant,
      this.gbrWithoutImplant,
      this.openSinusWithoutImplant,
      this.closedSinusWithoutImplant});

  TreatmentPlanSubModel.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    patientId = json['patientId'];
    patient = json['patient'] != null ? new DropDownDTO.fromJson(json['patient']) : null;
    tooth = json['tooth'];
    rootCanalTreatment = json['rootCanalTreatment'] != null ? new TreatmentPlanFieldsModel.fromJson(json['rootCanalTreatment']) : null;
    restoration = json['restoration'] != null ? new TreatmentPlanFieldsModel.fromJson(json['restoration']) : null;
    pontic = json['pontic'] != null ? new TreatmentPlanFieldsModel.fromJson(json['pontic']) : null;
    extraction = json['extraction'] != null ? new TreatmentPlanFieldsModel.fromJson(json['extraction']) : null;
    simpleImplant = json['simpleImplant'] != null ? new TreatmentPlanFieldsModel.fromJson(json['simpleImplant']) : null;
    immediateImplant = json['immediateImplant'] != null ? new TreatmentPlanFieldsModel.fromJson(json['immediateImplant']) : null;
    expansionWithImplant = json['expansionWithImplant'] != null ? new TreatmentPlanFieldsModel.fromJson(json['expansionWithImplant']) : null;
    splittingWithImplant = json['splittingWithImplant'] != null ? new TreatmentPlanFieldsModel.fromJson(json['splittingWithImplant']) : null;
    gbrWithImplant = json['gbrWithImplant'] != null ? new TreatmentPlanFieldsModel.fromJson(json['gbrWithImplant']) : null;
    openSinusWithImplant = json['openSinusWithImplant'] != null ? new TreatmentPlanFieldsModel.fromJson(json['openSinusWithImplant']) : null;
    closedSinusWithImplant = json['closedSinusWithImplant'] != null ? new TreatmentPlanFieldsModel.fromJson(json['closedSinusWithImplant']) : null;
    guidedImplant = json['guidedImplant'] != null ? new TreatmentPlanFieldsModel.fromJson(json['guidedImplant']) : null;
    expansionWithoutImplant = json['expansionWithoutImplant'] != null ? new TreatmentPlanFieldsModel.fromJson(json['expansionWithoutImplant']) : null;
    splittingWithoutImplant = json['splittingWithoutImplant'] != null ? new TreatmentPlanFieldsModel.fromJson(json['splittingWithoutImplant']) : null;
    gbrWithoutImplant = json['gbrWithoutImplant'] != null ? new TreatmentPlanFieldsModel.fromJson(json['gbrWithoutImplant']) : null;
    openSinusWithoutImplant = json['openSinusWithoutImplant'] != null ? new TreatmentPlanFieldsModel.fromJson(json['openSinusWithoutImplant']) : null;
    closedSinusWithoutImplant = json['closedSinusWithoutImplant'] != null ? new TreatmentPlanFieldsModel.fromJson(json['closedSinusWithoutImplant']) : null;
    scaling = json['scaling'] != null ? new TreatmentPlanFieldsModel.fromJson(json['scaling']) : null;
    crown = json['crown'] != null ? new TreatmentPlanFieldsModel.fromJson(json['crown']) : null;
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['patientId'] = this.patientId;
    data['tooth'] = this.tooth;
    if (this.rootCanalTreatment != null) {
      data['rootCanalTreatment'] = this.rootCanalTreatment!.toJson();
    }
    if (this.restoration != null) {
      data['restoration'] = this.restoration!.toJson();
    }
    if (this.pontic != null) {
      data['pontic'] = this.pontic!.toJson();
    }
    if (this.extraction != null) {
      data['extraction'] = this.extraction!.toJson();
    }
    if (this.simpleImplant != null) {
      data['simpleImplant'] = this.simpleImplant!.toJson();
    }
    if (this.immediateImplant != null) {
      data['immediateImplant'] = this.immediateImplant!.toJson();
    }
    if (this.expansionWithImplant != null) {
      data['expansionWithImplant'] = this.expansionWithImplant!.toJson();
    }
    if (this.splittingWithImplant != null) {
      data['splittingWithImplant'] = this.splittingWithImplant!.toJson();
    }
    if (this.gbrWithImplant != null) {
      data['gbrWithImplant'] = this.gbrWithImplant!.toJson();
    }
    if (this.openSinusWithImplant != null) {
      data['openSinusWithImplant'] = this.openSinusWithImplant!.toJson();
    }
    if (this.closedSinusWithImplant != null) {
      data['closedSinusWithImplant'] = this.closedSinusWithImplant!.toJson();
    }
    if (this.guidedImplant != null) {
      data['guidedImplant'] = this.guidedImplant!.toJson();
    }
    if (this.expansionWithoutImplant != null) {
      data['expansionWithoutImplant'] = this.expansionWithoutImplant!.toJson();
    }
    if (this.splittingWithoutImplant != null) {
      data['splittingWithoutImplant'] = this.splittingWithoutImplant!.toJson();
    }
    if (this.gbrWithoutImplant != null) {
      data['gbrWithoutImplant'] = this.gbrWithoutImplant!.toJson();
    }
    if (this.openSinusWithoutImplant != null) {
      data['openSinusWithoutImplant'] = this.openSinusWithoutImplant!.toJson();
    }
    if (this.closedSinusWithoutImplant != null) {
      data['closedSinusWithoutImplant'] = this.closedSinusWithoutImplant!.toJson();
    }
    if (this.scaling != null) {
      data['scaling'] = this.scaling!.toJson();
    }
    if (this.crown != null) {
      data['crown'] = this.crown!.toJson();
    }
    return data;
  }
}

class TreatmentPlanFieldsModel {
  String? value;
  bool? status;
  int? assignedToID;
  DropDownDTO? assignedTo;
  String? date;
  int? doneByAssistantID;
  DropDownDTO? doneByAssistant;
  int? doneBySupervisorID;
  DropDownDTO? doneBySupervisor;
  int? doneByCandidateID;
  DropDownDTO? doneByCandidate;
  int? doneByCandidateBatchID;
  DropDownDTO? doneByCandidateBatch;
  int? implantID;
  DropDownDTO? implant;
  int? price;
  int? planPrice;

  TreatmentPlanFieldsModel(
      {this.value = "",
      this.status = false,
      this.assignedToID,
      this.assignedTo,
      this.date,
      this.doneByAssistantID,
      this.doneByAssistant,
      this.doneBySupervisorID,
      this.doneBySupervisor,
      this.doneByCandidateID,
      this.doneByCandidate,
      this.doneByCandidateBatchID,
      this.doneByCandidateBatch,
      this.implantID,
        this.price=0,
        this.planPrice=0,
      this.implant}) {
    doneByAssistant = DropDownDTO();
    doneByCandidate = DropDownDTO();
    doneByCandidateBatch = DropDownDTO();
    doneBySupervisor = DropDownDTO();
    implant = DropDownDTO();
  }

  TreatmentPlanFieldsModel.fromJson(Map<String, dynamic> json) {
    value = json['value'] ?? "";
    status = json['status'] ?? false;
    assignedToID = json['assignedToID'];
    assignedTo = json['assignedTo'] != null ? new DropDownDTO.fromJson(json['assignedTo']) : null;
    date = json['date'];
    doneByAssistantID = json['doneByAssistantID'];
    doneByAssistant = json['doneByAssistant'] != null ? new DropDownDTO.fromJson(json['doneByAssistant']) : DropDownDTO();
    doneBySupervisorID = json['doneBySupervisorID'];
    doneBySupervisor = json['doneBySupervisor'] != null ? new DropDownDTO.fromJson(json['doneBySupervisor']) : DropDownDTO();
    doneByCandidateID = json['doneByCandidateID'];
    doneByCandidate = json['doneByCandidate'] != null ? new DropDownDTO.fromJson(json['doneByCandidate']) : DropDownDTO();
    doneByCandidateBatchID = json['doneByCandidateBatchID'];
    doneByCandidateBatch = json['doneByCandidateBatch'] != null ? new DropDownDTO.fromJson(json['doneByCandidateBatch']) : DropDownDTO();
    implantID = json['implantID'];
    price = json['price']??0;
    planPrice = json['planPrice'];
    implant = json['implant'] != null ? new DropDownDTO.fromJson(json['implant']) : DropDownDTO();
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['value'] = this.value;
    data['status'] = this.status;
    data['price'] = this.price;
    data['planPrice'] = this.planPrice??this.price;
    data['assignedToID'] = this.assignedToID;

    data['date'] =this.date!=null? this.date!.replaceFirst(" ", "T"):null;
    data['doneByAssistantID'] = this.doneByAssistantID;

    data['doneBySupervisorID'] = this.doneBySupervisorID;

    data['doneByCandidateID'] = this.doneByCandidateID;

    data['doneByCandidateBatchID'] = this.doneByCandidateBatchID;

    data['implantID'] = this.implantID;

    return data;
  }
}

File: cariro_implant_academy/Backend/CIA/Models/CIA/TreatmentModels/TreatmentPlanModel.cs
using CIA.Models.CIA;
using CIA.Models.CIA.TreatmentModels;
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace CIA.Models.TreatmentModels
{



    public class TreatmentPlanModel
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]

        public int? Id { get; set; }

        [ForeignKey("Patient")]
        public int? PatientId { get; set; }

        [ForeignKey("Operator.IdInt")]
        public int? OperatorId { get; set; }
        public ApplicationUser? Operator { get; set; }
        public DateTime? Date { get; set; }
        public bool ClearanceUpper { get; set; } = false;
        public bool ClearanceLower { get; set; } = false;

        public EnumWebsite Website { get; set; } = EnumWebsite.CIA;
        public class TreatmentPlanSubModel
        {
            [Key]
            [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
            public int? Id { get; set; }

            [ForeignKey("Patient")]
            public int? PatientId { get; set; }
            public Patient? Patient { get; set; }
            public int Tooth { get; set; }

            public TreatmentPlanFieldsModel? Scaling { get; set; }
            public TreatmentPlanFieldsModel? Crown { get; set; }
            public TreatmentPlanFieldsModel? RootCanalTreatment { get; set; }
            public TreatmentPlanFieldsModel? Restoration { get; set; }
            public TreatmentPlanFieldsModel? Pontic { get; set; }
            public TreatmentPlanFieldsModel? Extraction { get; set; }
            public TreatmentPlanFieldsModel? SimpleImplant { get; set; }
            public TreatmentPlanFieldsModel? ImmediateImplant { get; set; }
            public TreatmentPlanFieldsModel? ExpansionWithImplant { get; set; }
            public TreatmentPlanFieldsModel? SplittingWithImplant { get; set; }
            public TreatmentPlanFieldsModel? GBRWithImplant { get; set; }
            public TreatmentPlanFieldsModel? OpenSinusWithImplant { get; set; }
            public TreatmentPlanFieldsModel? ClosedSinusWithImplant { get; set; }
            public TreatmentPlanFieldsModel? GuidedImplant { get; set; }
            public TreatmentPlanFieldsModel? ExpansionWithoutImplant { get; set; }
            public TreatmentPlanFieldsModel? SplittingWithoutImplant { get; set; }
            public TreatmentPlanFieldsModel? GBRWithoutImplant { get; set; }
            public TreatmentPlanFieldsModel? OpenSinusWithoutImplant { get; set; }
            public TreatmentPlanFieldsModel? ClosedSinusWithoutImplant { get; set; }

            public EnumWebsite Website { get; set; } = EnumWebsite.CIA;



        }

        [Owned]
        public class TreatmentPlanFieldsModel
        {
            [ForeignKey("RequestChangeModel")]
            public int? RequestChangeId { get; set; }
            public RequestChangeModel? RequestChangeModel { get; set; }
            public String Value { get; set; } = "";
            public bool Status { get; set; }
            [ForeignKey("AssignedTo.IdInt")]
            public int? AssignedToID { get; set; }
            public int? PlanPrice { get; set; } = 0;

            public ApplicationUser? AssignedTo { get; set; }
            public DateTime? Date { get; set; }

            [ForeignKey("DoneByAssistant.IdInt")]
            public int? DoneByAssistantID { get; set; }
            public ApplicationUser? DoneByAssistant { get; set; }
            [ForeignKey("DoneBySupervisor.IdInt")]
            public int? DoneBySupervisorID { get; set; }
            public ApplicationUser? DoneBySupervisor { get; set; }
            [ForeignKey("DoneByCandidate.IdInt")]
            public int? DoneByCandidateID { get; set; }
            public ApplicationUser? DoneByCandidate { get; set; }
            [ForeignKey("CandidatesBatch")]
            public int? DoneByCandidateBatchID { get; set; }
            public DropDowns? DoneByCandidateBatch { get; set; }
            [ForeignKey("Implant")]
            public int? ImplantID { get; set; }
            public Implant? Implant { get; set; }
            [ForeignKey("ImplantRequest")]
            public int? ImplantIDRequest { get; set; }
            public Implant? ImplantRequest { get; set; }

            public EnumWebsite Website { get; set; } = EnumWebsite.CIA;
        }


    


}


    public class TreatmentDetailsModel
    {

        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int? Id { get; set; }

        [ForeignKey("PostSurgeryModelId")]
        public int? PostSurgeryModelId { get; set; }
        [ForeignKey("TreatmentItemModel")]
        public int? TreatmentItemId { get; set; }
        public TreatmentItemModel? TreatmentItem { get; set; }
        public int Tooth { get; set; }
        public String Name { get; set; } = "";

        [ForeignKey("Patient")]
        public int? PatientId { get; set; }
        public Patient? Patient { get; set; }
        public EnumWebsite Website { get; set; } = EnumWebsite.CIA;

        [ForeignKey("RequestChangeModel")]
        public int? RequestChangeId { get; set; }
        public RequestChangeModel? RequestChangeModel { get; set; }
        public String Value { get; set; } = "";
        public bool Status { get; set; }

        [ForeignKey("AssignedTo.IdInt")]
        public int? AssignedToID { get; set; }
        public ApplicationUser? AssignedTo { get; set; }

        public int? PlanPrice { get; set; } = 0;

        public DateTime? Date { get; set; }

        [ForeignKey("DoneByAssistant.IdInt")]
        public int? DoneByAssistantID { get; set; }
        public ApplicationUser? DoneByAssistant { get; set; }
        [ForeignKey("DoneBySupervisor.IdInt")]
        public int? DoneBySupervisorID { get; set; }
        public ApplicationUser? DoneBySupervisor { get; set; }
        [ForeignKey("DoneByCandidate.IdInt")]
        public int? DoneByCandidateID { get; set; }
        public ApplicationUser? DoneByCandidate { get; set; }
        [ForeignKey("CandidatesBatch")]
        public int? DoneByCandidateBatchID { get; set; }
        public DropDowns? DoneByCandidateBatch { get; set; }
        [ForeignKey("Implant")]
        public int? ImplantID { get; set; }
        public Implant? Implant { get; set; }
        [ForeignKey("ImplantRequest")]
        public int? ImplantIDRequest { get; set; }
        public Implant? ImplantRequest { get; set; }

    }

    public class PostSurgeryModel
    {

        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int? Id { get; set; }
        [ForeignKey("TreatmentDetailsModelId")]
        public int? TreatmentDetailsModelId { get; set; }

        [ForeignKey("Patient")]
        public int? PatientId { get; set; }
        public List<RequestChangeModel>? RequestChanges { get; set; }
        public bool? GuidedBoneRegeneration { get; set; }
        public bool? GuidedBoneRegeneration_BlockGraft { get; set; }
        public bool? GuidedBoneRegeneration_BlockGraft_Chin { get; set; }
        public bool? GuidedBoneRegeneration_BlockGraft_Ramus { get; set; }
        public bool? GuidedBoneRegeneration_BlockGraft_Tuberosity { get; set; }
        public string? GuidedBoneRegeneration_BlockGraft_Other { get; set; }
        public bool? GuidedBoneRegeneration_CutBy { get; set; }
        public bool? GuidedBoneRegeneration_CutBy_Disc { get; set; }
        public bool? GuidedBoneRegeneration_CutBy_Piezo { get; set; }
        public bool? GuidedBoneRegeneration_CutBy_Screws { get; set; }
        public int? GuidedBoneRegeneration_CutBy_ScrewsNumber { get; set; }
        public bool? GuidedBoneRegeneration_BoneParticle { get; set; }
        public int? GuidedBoneRegeneration_BoneParticle_100Autogenous { get; set; }
        public int? GuidedBoneRegeneration_BoneParticle_100Xenograft { get; set; }
        //public string? GuidedBoneRegeneration_BoneParticle_XenograftPercent { get; set; }
        public bool? GuidedBoneRegeneration_ACMBur { get; set; }
        public string? GuidedBoneRegeneration_ACMBur_Area { get; set; }
        public string? GuidedBoneRegeneration_ACMBur_Notes { get; set; }
        public bool? OpenSinusLift { get; set; }
        public bool? OpenSinusLift_Approach { get; set; }
        public string? OpenSinusLift_Approach_String { get; set; }
        public bool? OpenSinusLift_FillMaterial { get; set; }
        public string? OpenSinusLift_FillMaterial_String { get; set; }


        [ForeignKey("Membrane")]
        public int? OpenSinusLift_MembraneID { get; set; }
        public Membrane? OpenSinusLift_Membrane { get; set; }
        [ForeignKey("MembraneCompany")]
        public int? OpenSinusLift_Membrane_CompanyID { get; set; }
        public MembraneCompany? OpenSinusLift_Membrane_Company { get; set; }
        public int? OpenSinusLiftTacsNumber { get; set; }


        [ForeignKey("TacCompany")]
        public int? OpenSinusLift_TacsCompanyID { get; set; }
        public TacCompany? OpenSinusLift_TacsCompany { get; set; }
        public bool? SoftTissueGraft { get; set; }
        public bool? SoftTissueGraft_SurgeryType { get; set; }
        public bool? SoftTissueGraft_SurgeryType_SoftTissueGraft { get; set; }
        public bool? SoftTissueGraft_SurgeryType_Advanced { get; set; }
        public bool? SoftTissueGraft_SurgeryType_FreeGinivalGraft { get; set; }
        public bool? SoftTissueGraft_SurgeryType_ConnectiveTissueGraft { get; set; }
        public string? SoftTissueGraft_SurgeryType_SurgeryTechnique { get; set; }
        public bool? SoftTissueGraft_Exposure { get; set; }
        public string? SoftTissueGraft_Exposure_CustomizedHealingCollarTeethNumber { get; set; }
        public bool? SoftTissueGraft_DonorSite { get; set; }
        public string? SoftTissueGraft_DonorSite_Notes { get; set; }
        public bool? SoftTissueGraft_Suture { get; set; }
        public string? SoftTissueGraft_Suture_Material { get; set; }
        public string? SoftTissueGraft_Suture_Technique { get; set; }
        public string? SoftTissueGraft_Suture_PackType { get; set; }
        public bool? SoftTissueGraft_RecipientSite { get; set; }
        public string? SoftTissueGraft_RecipientSite_Area { get; set; }
        public bool? SoftTissueGraft_Augmentation { get; set; }
        public bool? SoftTissueGraft_Augmentation_Buccal { get; set; }
        public bool? SoftTissueGraft_Augmentation_Crestal { get; set; }
        public bool? SoftTissueGraft_Augmentation_Lingual { get; set; }
        public bool? SoftTissueGraft_Augmentation_Mesial { get; set; }
        public bool? SoftTissueGraft_Augmentation_Distal { get; set; }
        public bool? SoftTissueGraft_Frenectomy { get; set; }
        public string? SoftTissueGraft_Frenectomy_Notes { get; set; }
        public bool? SoftTissueGraft_BoneGraft { get; set; }
        public string? SoftTissueGraft_BoneGraft_Notes { get; set; }
        public bool? SutureAndTemporizationAndXRay { get; set; }
        public bool? SutureAndTemporizationAndXRay_SutureSize { get; set; }
        public bool? SutureAndTemporizationAndXRay_SutureSize_3_0 { get; set; }
        public bool? SutureAndTemporizationAndXRay_SutureSize_4_0 { get; set; }
        public bool? SutureAndTemporizationAndXRay_SutureSize_5_0 { get; set; }
        public bool? SutureAndTemporizationAndXRay_SutureSize_6_0 { get; set; }
        public bool? SutureAndTemporizationAndXRay_SutureSize_7_0 { get; set; }
        public bool? SutureAndTemporizationAndXRay_SutureSize_ImplantSubcrestal { get; set; }
        public bool? SutureAndTemporizationAndXRay_Material { get; set; }
        public bool? SutureAndTemporizationAndXRay_Material_Vicryl { get; set; }
        public bool? SutureAndTemporizationAndXRay_Material_Proline { get; set; }
        public bool? SutureAndTemporizationAndXRay_Material_XRay { get; set; }
        public string? SutureAndTemporizationAndXRay_Material_SutureTechnique { get; set; }
        public bool? SutureAndTemporizationAndXRay_Temporary { get; set; }
        public bool? SutureAndTemporizationAndXRay_Temporary_HealingCollar { get; set; }
        public bool? SutureAndTemporizationAndXRay_Temporary_CustomizedHeallingCollar { get; set; }
        public bool? SutureAndTemporizationAndXRay_Temporary_Crown { get; set; }
        public bool? SutureAndTemporizationAndXRay_Temporary_MarylandBridge { get; set; }
        public bool? SutureAndTemporizationAndXRay_Temporary_BridgeOnTeeth { get; set; }
        public bool? SutureAndTemporizationAndXRay_Temporary_DentureWithGlassFiber { get; set; }

    }

    public class TreatmentItemModel
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int? Id { get; set; }
        public String Name { get; set; }
        public int Price { get; set; } = 0;
        public String? PriceAction { get; set; }
        public bool AllowAssign { get; set; } = true;
        public bool ShowInSurgical { get; set; } = true;
        public EnumWebsite Website { get; set; } = EnumWebsite.CIA;
        
    }

}
